components/MultiInsert/MultiInsert.tsx

```diff
@@ -0,0 +1,329 @@
+      descriptionInputRef: createRef<HTMLTextAreaElement>(),
+      nameError: false,
+      addressError: false,
+      removeButtonRef: createRef<HTMLButtonElement>(),
+    };
+    setWebsites((current) => [...current, newWebsite]);
+    setMultiInsertItemID(multiInsertItemID + 1);
+    setFormInvalid(true);
+  };
+  const removeWebsite = (id: number) => {
+    const indexOfRemovedItem = websites.findIndex(
+      (website) => website.id === id,
+    );
+    // If there is only one item in the array, set focus to the add new button
+    if (websites.length === 1) {
+      addNewButtonRef.current?.focus();
+      // Else, if the removed item was the last item, set focus to the previous one
+    } else if (indexOfRemovedItem === websites.length - 1) {
+      websites[indexOfRemovedItem - 1].nameInputRef.current?.focus();
+      // Else  set focus to the next element
+    } else {
+      websites[indexOfRemovedItem + 1].nameInputRef.current?.focus();
+    }
+
+    // Removed websites are preferably announced with their title
+    // If title is not present, pass the ID of the element to the list of toasts
+    let toastText = websites[indexOfRemovedItem].nameInputRef.current?.value;
+    if (!toastText) {
+      toastText = String(websites[indexOfRemovedItem].id);
+    }
+    flashToast(toastText, websites[indexOfRemovedItem].id);
+    setWebsites(websites.filter((website) => website.id !== id));
+  };
+
+  const toggleInputErrorState = (
+    status: boolean,
+    id: number,
+    inputField: 'name' | 'address',
+  ) => {
+    switch (inputField) {
+      case 'name':
+        setWebsites(
+          websites.map((website) =>
+            website.id === id ? { ...website, nameError: status } : website,
+          ),
+        );
+        break;
+      case 'address':
+        setWebsites(
+          websites.map((website) =>
+            website.id === id ? { ...website, addressError: status } : website,
+          ),
+        );
+        break;
+      default:
+        break;
+    }
+  };
+
+  const checkWebsitesValidity = () => {
+    setFormInvalid(
+      websites.some(
+        (website) =>
+          website.nameInputRef.current?.value === '' ||
+          website.addressInputRef.current?.value === '',
+      ),
+    );
+  };
+
+  const flashToast = (toastName: string, id: number) => {
+    setToasts((current) => [...current, { text: toastName, id: id }]);
+    setTimeout(() => {
+      setToasts((current) => {
+        return current.filter((t) => t.id !== id);
+      });
+    }, 5000);
+  };
+
+  const save = () => {
+    // Remove possible empty website entries
+    const validWebsites = websites.filter(
+      (website) =>
+        website.nameInputRef.current?.value !== '' ||
+        website.addressInputRef.current?.value !== '' ||
+        website.descriptionInputRef.current?.value !== '',
+    );
+    if (validWebsites.length === 0) {
+      alert('Lisää vähintään yksi verkkosivu!');
+      return;
+    }
+
+    console.log(
```

Comment:
Is console log left from testing or wanted in the multi insert pattern demo?

Reply:
It is a wanted thing in this instance. The "save" button does not have any other functionality so I thought it could console log the website inputs